llama_index/llms/monsterapi.py

```diff
@@ -0,0 +1,119 @@
+from typing import Any, Callable, Dict, Optional, Sequence
+from pydantic import Field, PrivateAttr
+
+
+from llama_index.callbacks import CallbackManager
+from llama_index.constants import DEFAULT_CONTEXT_WINDOW, DEFAULT_NUM_OUTPUTS
+from llama_index.llms.base import (
+    ChatMessage,
+    ChatResponse,
+    CompletionResponse,
+    CompletionResponseGen,
+    LLMMetadata,
+    llm_chat_callback,
+    llm_completion_callback
+)
+from llama_index.llms.custom import CustomLLM
+from llama_index.llms.generic_utils import \
+    messages_to_prompt as generic_messages_to_prompt
+
+
+class MonsterLLM(CustomLLM):
+
+    model: str = Field(description="The Predibase model to use.")
+    monster_api_key: Optional[str] = Field(
+        description="The Predibase API key to use.")
+    max_new_tokens: int = Field(
+        description="The number of tokens to generate.")
+    temperature: float = Field(
+        description="The temperature to use for sampling.")
+    context_window: int = Field(
+        description="The number of context tokens available to the LLM."
+    )
+    messages_to_prompt: Optional[Callable] = None
+
+    _client: Any = PrivateAttr()
+
+    def __init__(
+        self,
+        model: str,
+        monster_api_key: Optional[str] = None,
+        max_new_tokens: int = 256,
```

Comment:
nit: we can use the `DEFAULT_NUM_OUTPUTS` constant here

Reply:
True, Updated it!